numpy/core/src/multiarray/buffer.c

```diff
@@ -833,13 +862,34 @@ gentype_getbuffer(PyObject *self, Py_buffer *view, int flags)
     }
 #endif
     view->len = elsize;
+    if (PyArray_IsScalar(self, Datetime) || PyArray_IsScalar(self, Timedelta)) {
+        elsize = 1; /* descr->elsize is 8, but scalars are always 1 */
```

Comment:
This comment should be explaining that the format is being exposed as `B` - `scalars are always 1` is false - `sizeof(B) == 1` is what's relevant here.

Reply:
removing code, comment expanded above.

Reply:
Whoops, this is needed. Comment fixed.